libs/math/matrix_tools.h

```diff
@@ -358,6 +365,16 @@ matrix_get_diagonal (Matrix<T,N,N> const& mat)
     return diag;
 }
 
+template <typename T, int N>
+inline T
+matrix_trace(math::Matrix<T, N, N> const& mat)
+{
+    T ret = T(0.0);
+    for (int i = 0; i < N; ++i)
+        ret += mat(i, i);
```

Comment:
Instead of using operator(), I think this can be made more efficient using operator[] and adding N+1 in the for increment.